### ISSUE TITLE
[Suggestion for Firefox] I used to disable the trash folder to hide the "Other Bookmarks" folder. Shouldn't there be a way to let users know that they can hide it without disabling it? Like a landing page?

### ISSUE THREAD

**Issue by YousufSSyed** [aaFn/Bookmark-search-plus-2#237]
The trash folder appears in the "Other Bookmarks" folder, which is off to the side in the bookmarks toolbar, but I don't like having it there because that takes away room for my other bookmarks. Ever since the trash folder has been added to Bookmarks Search Plus 2, I would always disable it to the hide the folder. Today I found out I can hide the Other Bookmarks without disable the trash folder: by right clicking on "Other Bookmarks" and unchecking "Show Other Bookmarks."

I think there should be a way to let users know to do this, perhaps with an extension landing page. A landing page would also be good for explaining the the extension and its features.

**Comment by aaFn** [comment]
Hello @YousufSSyed , not sure  how you would like people to get to the landing page ?
Could you illustrate a little more in details by the typical series of user action you would like to see happening for the user to get to it ?

I guess the landing page itself could be either https://addons.mozilla.org/en-US/firefox/addon/bookmark-search-plus-2/ or https://github.com/aaFn/Bookmark-search-plus-2/wiki ..

**Comment by Gitoffthelawn** [comment]
Just my personal preference, but I don't like landing pages.  I find them to be intrusive.  What I do like, however, is a link in the description (and settings page, if the extension has one) for additional help.  Ideally, the link is to a page on a major service like readthedocs.io, GitLab, or GitHub (but nothing owned by Google!), and not a private website.

**Comment by YousufSSyed** [comment]
Documentation for the extension like that would be nice. 

The user wouldn't do anything to get to the landing page, I should've more specific. I don't know how it would be programmed into an extension, but I know that some extensions in the Firefox Addons store open a page as soon as you install the extension (if you ever logged onto an account on a web browser and had it sync all of your extensions, you might notice extension pages suddenly opening up). That's what I had in mind.

It might be intrusive? Such landing pages show once, and only once. If a user didn't want to read the page, they could just close the tab.

**Comment by Gitoffthelawn** [comment]
@YousufSSyed Yes, that's the typical definition of a landing page.

Although I'm all in favour of including a link for the user to manually open such a page, ***forced*** landing pages have several drawbacks:  
1. Workflow interruption - if the user has extensions set to auto-update, suddenly (and without notice) a new browser tab will appear.  If the user has multiple extensions with updates, and each forces a landing page, multiple new tabs will suddenly appear, which results in bad UX.
2. Security - landing pages are prime targets for hackers/phishers.  Hackers/phishers know that the page (including scripts on it) will be loaded automatically by thousands of unsuspecting people.
3. Unwanted connections - many people do not want their computer (including web browsers) making connections that the user has not initiated by choice.  In some settings, where all connections are logged, connections by each user are closely monitored and any connections to unapproved addresses will come under scrutiny.

For these reasons (plus several other less prominent reasons), I recommend never force-loading any landing pages.  Providing link(s) to documentation/tips/changelogs/support is typically a more advisable solution.

**Comment by YousufSSyed** [comment]
1. Extension landing pages are shown once, right after installing and the extension and not every time the extension is updated.
2. (and 3) Extension page URLs (including landing pages) use the moz-extension:// protocol, The [Internet Archive Extension](https://addons.mozilla.org/en-US/firefox/addon/wayback-machine_new/) has a landing with an agreement which activates when you first use the extension (like when you click on it the 1st time). Some extensions, like [Tab Session Manager](https://addons.mozilla.org/en-US/firefox/addon/tab-session-manager/) have a settings page. When you open it, it takes you to a page with said protocol, even when Wi-Fi is turned off. But I guess you would need to do more research to see how the 

But at the end of the day, it's up to @aaFn if they want to add a landing page, or a link with extension documentation.

**Comment by YousufSSyed** [comment]
Honestly, I think GitOffTheLawn is exaggerating the downsides of a landing page, from what I can tell they're nonexistant. Many extension in the Firefox Addons store do it, heck @aaFn don't u have any extensions that do so? Do you think it would be a problem if Bookmarks Search Plus 2 had one too?

**Comment by Gitoffthelawn** [comment]
@YousufSSyed Sorry I haven't had a chance to reply.  I thought you were referring to an external "landing page".  A static landing page delivered through the `moz-extension` protocol resolves the second and third concerns, while only leaving the first concern (workflow interruption).

Thinking more about it, though, a readily available link is often preferable to a "landing page" because it will never interrupt workflow, plus has the advantage of beings available at all times (eliminating the user wondering "how do I find that helpful page again?").  That said, a static "landing page" that opens via the `moz-extension` protocol will likely be, at worst,  just a minor annoyance.

**Comment by aaFn** [comment]
Hello gents, what do you mean by "a landing page using the moz-extension:// protocol," ?
Do you mean a page shipped as part of the add-on package (inside it), and so a local page as seen from the browser ?

**Comment by Gitoffthelawn** [comment]
> ...what do you mean by "a landing page using the moz-extension:// protocol," ? Do you mean a page shipped as part of the add-on package (inside it), and so a local page as seen from the browser ?

Yes. BTW, if you choose to do this, you can create a link in the UI to open such a page.

**Comment by aaFn** [comment]
Hello, well, all the documentation is in the wiki on https://github.com/aaFn/Bookmark-search-plus-2/wiki
and I am not inclined to duplicate something in the add-on itself, so not sure what to do on this topic.

I could add maybe an internal page in the add-on activated on installation only and linking to the wiki,, however I am not sure of the value of that ...

**Comment by YousufSSyed** [comment]
@aaFn You could make a brief landing page linking the wiki and adding a brief note about the trash folder.
